### ISSUE TITLE
Console error "React keys must be passed directly to JSX without using spread"

### ISSUE THREAD

**Issue by wimbarelds** [pegasystems/react-sdk-components#389]
**Describe the bug**
An error that frequently pops into the (chrome) console is:
```
Warning: A props object containing a "key" prop is being spread into JSX:
  let props = {key: someKey, readOnly: ..., type: ..., name: ..., context: ..., getPConnect: ..., additionalProps: ..., validatemessage: ..., children: ...};
  <Connect(Reference2) {...props} />
React keys must be passed directly to JSX without using spread:
  let props = {readOnly: ..., type: ..., name: ..., context: ..., getPConnect: ..., additionalProps: ..., validatemessage: ..., children: ...};
  <Connect(Reference2) key={someKey} {...props} />
```

**To Reproduce**
1. Go to any page using a reference/View rendered using PConnectComponent (in SDK-Portal?)
2. See error

**Expected behavior**
No error.

**Screenshots**
<img width="990" alt="image" src="https://github.com/user-attachments/assets/bcf629fc-0081-4414-8720-2bc25bd509a0">

**Additional context**

The error is causes by:

`https://github.com/pegasystems/react-sdk-components/blob/master/packages/react-sdk-components/src/bridge/react_pconnect.jsx:284-290`

Current code:

```tsx
    // If the new component is a reference node then mark with a unique key
    if (['reference', 'View'].includes(getPConnect().getComponentName()) && !finalProps.key) {
      finalProps.key = this.getKey();
    }

    // console.log(`react_pconnect: used to return: <this.Control {...finalProps} />`);

    return <this.Control {...finalProps}>{this.createChildren()}</this.Control>;
```

Code that fixes the error:

```tsx
    // If the new component is a reference node then mark with a unique key
    if (['reference', 'View'].includes(getPConnect().getComponentName()) && !finalProps.key) {
      return <this.Control {...finalProps} key={this.getKey()}>{this.createChildren()}</this.Control>;
    }

    // console.log(`react_pconnect: used to return: <this.Control {...finalProps} />`);

    return <this.Control {...finalProps}>{this.createChildren()}</this.Control>;
```

**Comment by tumms2021389** [comment]
Thank you, @wimbarelds , for bringing up the issue. We'll investigate it and add a fix for it.

**Comment by tumms2021389** [comment]
@wimbarelds , this issue has been fixed and will be part of 24.2.10 version
